exams/pearson/audit.py

```diff
@@ -0,0 +1,265 @@
+
+        Args:
+            filename (str):
+                filename of the encrypted file
+            file_type (str):
+                type of the encrypted file
+
+        Returns:
+            boto.s3.key.Key:
+                the key to store the file in
+        """
+        _, filename = os.path.split(filename)
+        return key.Key(
+            self.get_bucket(),
+            'exam_audits/{file_type}/{filename}'.format(
+                filename=filename,
+                file_type=file_type,
+            )
+        )
+
+    def upload(self, filename, file_type):
+        """
+        Determines the S3 key to store the file under
+
+        Args:
+            filename (str):
+                filename of the encrypted file
+            file_type (str):
+                type of the encrypted file
+
+        Returns:
+            str:
+                the key path in S3 where the file way stored
+        """
+        s3_key = self.get_s3_key(filename, file_type)
+        s3_key.set_contents_from_filename(
+            filename,
+            headers={
+                'x-amz-server-side-encryption': SSE_ENCRYPTION_ALGORITHM,
+            }
+        )
+        return s3_key.key
+
+
+class ExamDataAuditor:
+    """
+    Encrypted file auditor for exam requests/responses
+    """
+    REQUEST = 'request'
+    RESPONSE = 'response'
+
+    def __init__(self, gpg=None, store=None):
+        self.gpg = gpg or gnupg.GPG()
+        self.store = store or S3AuditStorage()
+
+    @property
+    def fingerprint(self):
+        """The fingerprint for the public GPG key"""
+        return settings.EXAMS_AUDIT_ENCRYPTION_FINGERPRINT
+
+    @property
+    def public_key(self):
+        """The public GPG key"""
+        # unescape newlines (from an env var)
+        return re.sub(r'\\\\n', '\n', settings.EXAMS_AUDIT_ENCRYPTION_PUBLIC_KEY)
+
+    def get_fingerprints(self):
+        """
+        Get the fingerprints of keys currently loaded on the system
+
+        Returns:
+            list:
+                list of keys
+        """
+        return self.gpg.list_keys().fingerprints
+
+    def configure(self):
+        """
+        Performs a one-time configuration of GPG keys
+
+        Raises:
+            ImproperlyConfigured:
+                settings are missing or invalid
+
+        Returns:
+            bool:
+                True if configuration successful
+        """
+        # if any of the settings are not set, we consider this feature not enabled
+        if any(getattr(settings, key) is None for key in REQUIRED_SETTINGS):
+            raise ImproperlyConfigured(
+                "One of EXAMS_AUDIT_ENCRYPTION_PUBLIC_KEY or "
+                "EXAMS_AUDIT_ENCRYPTION_FINGERPRINT is required but not set"
+            )
+
+        if self.fingerprint in self.get_fingerprints():
+            return
+
+        result = self.gpg.import_keys(self.public_key)
+
+        counts = dict(result.counts)
+
+        if counts['not_imported'] != 0:
```

Comment:
That method doesn't raise an error (it probably does raise low-level IO errors), instead it returns a [`gnupg._parsers.ImportResult`](https://github.com/isislovecruft/python-gnupg/blob/master/gnupg/_parsers.py#L1087). I can't speak to the author's design decisions, since it's calling the `gpg` CLI, they probably though it best to return a parsed response of the command output rather than erroring.

I honestly can't defend the cast of `results.count` (an `OrderedDict`) to a `dict`, so that can be removed.